packages/amplify_datastore/test/amplify_datastore_delete_test.dart

```diff
@@ -0,0 +1,66 @@
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:amplify_datastore/amplify_datastore.dart';
+import 'package:flutter/services.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+import './testData/Post.dart';
+
+void main() {
+  const MethodChannel dataStoreChannel =
+      MethodChannel('com.amazonaws.amplify/datastore');
+
+  AmplifyDataStore dataStore = AmplifyDataStore(modelSchemas: null);
+
+  TestWidgetsFlutterBinding.ensureInitialized();
+
+  setUp(() {});
+
+  tearDown(() {
+    dataStoreChannel.setMockMethodCallHandler(null);
+  });
+
+  dynamic getJsonFromFile(String path) async {
+    path = 'resources/delete_api/' + path;
+    String jsonString = '';
+    try {
+      jsonString = await File(path).readAsString();
+    } catch (e) {
+      jsonString = await File('test/' + path).readAsString();
+    }
+    return jsonDecode(jsonString);
+  }
+
+  test('delete returns 1 sucessful result (no query predicate)', () async {
+
+    var json = await getJsonFromFile('response/1_deleted_result.json');
+
+    dataStoreChannel.setMockMethodCallHandler((MethodCall methodCall) async {
+      if (methodCall.method == "delete") {
+        return json;
+      }
+    });
+
+    Post instance = Post(title: json["title"], rating: json["rating"], created: json["created"], id: json["id"]);
+
+    Post post = await dataStore.delete(instance);
+    expect(post.id, instance.id);
+  });
+
+    test('delete returns 1 sucessful result (with query predicate)', () async {
```

Comment:
The indentation seems off here

Reply:
Fixed.

Reply:
You can enable format on save. The Flutter/Dart plugins have this included, and can save you from having to worry about indentation.